lazy_load_symbols/load_sym_lazily.py

```diff
@@ -0,0 +1,255 @@
+
+class LoadLazySymbols(gdb.Command):
+    '''
+    A command to load a recording without loading
+    the symbols and then loading only the required symbols.
+    '''
+
+    def __init__(self):
+        super(LoadLazySymbols, self).__init__('loadlazy', gdb.COMMAND_DATA, gdb.COMPLETE_FILENAME)
+        self.sodict = {}
+        self.rec = None
+
+
+    def load_all_symbols(self):
+        for content in self.sodict.values():
+            fn = content.fullname
+            content.symbols = extract_symbol_list_from_so(fn)
+
+    def invoke(self, rec, from_tty):
+        try:
+            if self.sodict and \
+               rec == self.rec:
+                pass
+            else:
+                self.rec = rec
+                gdb.execute('set auto-solib-add off', to_string=True)
+                gdb.execute('uload {}'.format(rec))
+        except Exception:
+            traceback.print_exc()
+
+class PopulateBT(gdb.Command):
+    '''
+    A command to read the symbols for an empty backtrace
+    '''
+
+    def __init__(self):
+        super(PopulateBT, self).__init__('popbt', gdb.COMPLETE_EXPRESSION)
+
+    def get_frame_name(pc):
+        sodict = load_so_list()
+        for name in sodict:
+            sa = int(sodict[name].start, 16)
+            ea = int(sodict[name].end, 16)
+            if pc >= sa and pc < ea:
+                gdb.execute('sharedlibrary {}'.format(name))
+
+    def invoke(self, arg, from_tty):
+        curr_frame = gdb.newest_frame()
+        frames_loaded = False
+        while curr_frame:
+            try:
+                # iteratively go through all frames and for each one check if the name is loaded.
+                # if not load the name
+                if not curr_frame.name():
+                    self.get_frame_name(curr_frame.pc())
+                    frames_loaded = True
+                curr_frame = curr_frame.older()
+            except gdb.error:
+                # restart as loading frames might change things ;)
+                if not frames_loaded:
+                    # Didn't load anything last time, let's break as we might be
+                    # in an infinite loop.
+                    break
+                else:
+                    frames_loaded = False
+                curr_frame = gdb.newest_frame()
+
+class ReverseStepSymbol(gdb.Command):
+    '''
+    A command to find reverse step and find symbols for the current program counter
+    '''
+
+    def __init__(self):
+        super(ReverseStepSymbol, self).__init__('rss', gdb.COMPLETE_EXPRESSION)
+
+    def get_frame_name(self, pc):
+        sodict = load_so_list()
+        loaded = False
+        try:
+            for name in sodict:
+                sa = int(sodict[name].start, 16)
+                ea = int(sodict[name].end, 16)
+                if pc >= sa and pc < ea:
+                    gdb.execute('sharedlibrary {}'.format(name))
+                    print('found sharedlibrary', name)
+                    loaded = True
```

Comment:
This is almost identical to `PopulateBT.get_frame_name` you could have a function which tries loading the library and returns a bool.